APE18.rst

```diff
@@ -0,0 +1,65 @@
+Adopt NEP 29 for CPython and Numpy Version Support
+--------------------------------------------------
+
+author: Stuart Mumford and Thomas Robitaille
+
+date-created: 2020 October 16
+
+date-last-revised: 2020 October 16
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+APE 10 previously determined supported CPython versions, and provided a list up to the 4.0 release.
+Since APE 10 was written the wider scientific Python community have written and largely adopted `NEP 29`_ as a policy for supported versions of both CPython and Numpy.
+This APE proposes that Astropy adopt this policy following other projects in the scientific ecosystem.
+
+
+Detailed description
+--------------------
+
+This APE fully adopts `NEP 29`_ for both CPython and Numpy versions for the core astropy package, and it supersedes APE 10 wherever there is a conflict between that APE and this APE or `NEP 29`_.
+
+While this APE will not replicate the whole of `NEP 29`_, and `NEP 29`_ should be considered the canonical source for the support policy, the core scheme to be adopted is as follows:
+
+The core astropy package will support:
+
+* All minor versions of Python released 42 months prior to the project, and at minimum the two latest minor versions.
+* All minor versions of numpy released in the 24 months prior to the project, and at minimum the last three minor versions.
+
```

Comment:
```suggestion
* All minor versions of Python released 42 months prior to a non-bugfix, and at minimum the two latest minor versions.
* All minor versions of numpy released in the 24 months prior to a non-bugfix, and at minimum the last three minor versions.
* Versions of other runtime dependencies released 24 prior to a non-bugfix release.

These versions of the dependencies are kept supported for bugfix lifetime of the given astropy version.

```

Reply:
So, this would introduce a policy for scipy/matplotlib/etc. We should not drop them to support only latest greatest, but then should linger with an old version neither. (inspired by @pllim seeking for input whether to drop 0.18 scipy)

Reply:
I'd be up for adding such a clause, though we should make it clear this is an addition compared to NEP 29?

Reply:
Yes, I'm happy to add a blurb about that if this is a go ahead.

Reply:
Scipy sticks to NEP 29 itself, so we could state that in general we try to main synchronism, i.e., pick that scipy that supports all of the numpy/python we support, but not more.

Reply:
I like to say we guarantee to support x versions, rather that we stay in sync, e.g. if there is no reason popping up to drop I would not drop a version (e.g., we could support very old scipies for a long time), but I would not put any extra effort in keeping up supporting an old versions when it's outside of the range.

Reply:
@bsipocz - agreed there is no reason to explicitly fail on older versions (as we do for numpy); at some level it is more about what version we still test -- and for which we thus *guarantee* support.

Reply:
The oldest supported one we should test with, anything older would not work as those versions are set in package metadata. I would only say we don't bump the versions unless any issues came up with them, e.g. now many versions turn out to be not python3.7 compatible, so we need to bump them along with dropping support for 3.6.

Reply:
I am :+1: on this suggestion